rosa_tutorials/rosa-mobb-verify-permissions-sts-deployment.adoc

```diff
@@ -0,0 +1,117 @@
+
+Details about the IAM resources required for an STS-enabled installation of ROSA can be found here: xref:../rosa_architecture/rosa-sts-about-iam-resources.adoc#rosa-sts-about-iam-resources[About IAM resources for ROSA clusters that use STS]
+
+This guide is validated for ROSA v4.11.X.
+
+== Prerequisites
+
+* link:https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html[AWS CLI]
+* xref:../cli_reference/rosa_cli/rosa-get-started-cli.adoc#rosa-get-started-cli[ROSA CLI] v1.2.6
+* link:https://stedolan.github.io/jq/[jq CLI]
+* link:https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_testing-policies.html[AWS role with required permissions]
+
+[id="verify-ROSA-permissions_{context}"]
+== Verifying ROSA permissions
+To verify the permissions required for ROSA we can run the script below without ever creating any AWS resources.
+
+The script uses the `rosa`, `aws`, and `jq` CLI commands to create files in the working directory that will be used to verify permissions in the account connected to the current AWS configuration.
+
+The AWS Policy Simulator is used to verify the permissions of each role policy against the API calls extracted by `jq`; results are then stored in a text file appended with `.results`.
+
+This script will verify the permissions for the current account and region:
+
+[source,terminal]
+----
+#!/bin/bash
+
+$ while getopts 'p:' OPTION; do
+  case "$OPTION" in
+    p)
+      PREFIX="$OPTARG"
+      ;;
+    ?)
+      echo "script usage: $(basename \$0) [-p PREFIX]" >&2
+      exit 1
+      ;;
+  esac
+done
+shift "$(($OPTIND -1))"
+
+$ rosa create account-roles --mode manual --prefix $PREFIX
+
+$ INSTALLER_POLICY=$(cat sts_installer_permission_policy.json | jq )
+$ CONTROL_PLANE_POLICY=$(cat sts_instance_controlplane_permission_policy.json | jq)
+$ WORKER_POLICY=$(cat sts_instance_worker_permission_policy.json | jq)
+$ SUPPORT_POLICY=$(cat sts_support_permission_policy.json | jq)
+
+$ simulatePolicy () {
+    outputFile="${2}.results"
+    echo $2
+    aws iam simulate-custom-policy --policy-input-list "$1" --action-names $(jq '.Statement | map(select(.Effect == "Allow"))[].Action | if type == "string" then . else .[] end' "$2" -r) --output text > $outputFile
+}
+
+$ simulatePolicy "$INSTALLER_POLICY" "sts_installer_permission_policy.json"
+$ simulatePolicy "$CONTROL_PLANE_POLICY" "sts_instance_controlplane_permission_policy.json"
+$ simulatePolicy "$WORKER_POLICY" "sts_instance_worker_permission_policy.json"
+$ simulatePolicy "$SUPPORT_POLICY" "sts_support_permission_policy.json"
+----
+
+[id="usage-instructions_{context}"]
+== Usage Instructions
+
+. To use the script, run the following commands in a `bash` terminal (the -p option defines a prefix for the roles):
++
+[source,terminal]
+----
+$ mkdir scratch
+$ cd scratch
+$ curl https://raw.githubusercontent.com/rh-mobb/documentation/main/content/docs/rosa/verify-permissions/verify-permissions.sh --output verify-permissions.sh
```

Comment:
Let's change this from pulling the file down from the internet and instead run the following command to create the file from the buffer itself: 

```
cat << 'EOF' > verify-permissions.sh
#!/bin/bash
while getopts 'p:' OPTION; do
  case "$OPTION" in
    p)
      PREFIX="$OPTARG"
      ;;
    ?)
      echo "script usage: $(basename \$0) [-p PREFIX]" >&2
      exit 1
      ;;
  esac
done
shift "$(($OPTIND -1))"
rosa create account-roles --mode manual --prefix $PREFIX
INSTALLER_POLICY=$(cat sts_installer_permission_policy.json | jq )
CONTROL_PLANE_POLICY=$(cat sts_instance_controlplane_permission_policy.json | jq)
WORKER_POLICY=$(cat sts_instance_worker_permission_policy.json | jq)
SUPPORT_POLICY=$(cat sts_support_permission_policy.json | jq)
simulatePolicy () {
    outputFile="${2}.results"
    echo $2
    aws iam simulate-custom-policy --policy-input-list "$1" --action-names $(jq '.Statement | map(select(.Effect == "Allow"))[].Action | if type == "string" then . else .[] end' "$2" -r) --output text > $outputFile
}
simulatePolicy "$INSTALLER_POLICY" "sts_installer_permission_policy.json"
simulatePolicy "$CONTROL_PLANE_POLICY" "sts_instance_controlplane_permission_policy.json"
simulatePolicy "$WORKER_POLICY" "sts_instance_worker_permission_policy.json"
simulatePolicy "$SUPPORT_POLICY" "sts_support_permission_policy.json"
EOF
```

Reply:
Gotcha, I'll replace line 89 with this entire command, let me know if I interpreted this incorrectly.